### PR TITLE
Fix issues on ARM builds

### LINKED ISSUE #2134: bugfixTests failed on Raspbian buster 10.10 (gcc 8.3.0)
##### **Describe the bug**
I followed the install process described in README and couldn't go through `ctest --verbose` command.
The objective is to install exiv2 on Raspbian Buster 10.10 running a gcc 8.3.0.
Tests failed (see [LastTest.log](https://github.com/Exiv2/exiv2/files/8206706/LastTest.log)):

> 1: FAIL: test_run (github.test_issue_1503.issue_1503_jxl_bmff)
> 1: FAIL: test_run (github.test_pr1475_AVIF.pr_1475_avif_avif)
> 1: FAIL: test_run (github.test_pr1475_AVIF.pr_1475_exif_xmp_avif)
> 1: FAIL: test_run (github.test_pr1475_AVIF.pr_1475_metadata2_avif)
> 1: FAIL: test_run (github.test_pr1475_HEIC.pr_1475_2021_heic)
> 1: FAIL: test_run (github.test_pr1475_HEIC.pr_1475_IMG_3578_heic)
> 1: FAIL: test_run (github.test_pr1475_HEIC.pr_1475_Stonehenge_heic)
> 1: FAIL: test_run (github.test_pr1475_HEIC.pr_1475_heic_heic)
> 1: FAIL: test_run (github.test_pr1475_HIF.pr_1475_Canon_hif)
> 1: FAIL: test_run (github.test_pr1475_HIF.pr_1475_Sony_hif)
> 1: FAILED (failures=10, skipped=10)

##### **To Reproduce**
HEAD of Branch `main` Raspbian Buster 10.10 GCC 8.3.0

Commands used :

```
git clone https://github.com/Exiv2/exiv2.git
cd exiv2
mkdir build && cd build
cmake -DCMAKE_BUILD_TYPE=Release ..
cmake --build .
ctest --verbose
```

##### **Expected behavior**
All tests passed without any failures.

##### **Desktop (please complete the following information):**
- Raspbian Buster 10.10 (Armv7l  i.e. 32 bit)
- GCC 8.3.0
- Release build : -DCMAKE_BUILD_TYPE=Release

##### **Additional context**
Installation goes perfectly right on Ubuntu 20.04 (x86_64 i.e. 64 bit) with GCC 9.4.0 and same install steps

### PR DESCRIPTION
**Pull request by piponazo**
Fix #2134 

I configured and built the project on my personal raspberry Pi and I found out few issues that I fix in this PR. 

I will try to add in the future a new job to build ARM builds (see [link](https://www.ryanwinter.org/creating-a-github-workflow-for-the-gnu-arm-compiler/)). Note: I could not find a github workflow environments for ARM hardware.

### CODE REVIEW

File: src/bmffimage.cpp [hunk 1]
```diff
@@ -184,7 +184,7 @@ long BmffImage::boxHandler(std::ostream& out /* = std::cout*/, Exiv2::PrintStruc
   if (bTrace) {
     bLF = true;
     out << indent(depth) << "Exiv2::BmffImage::boxHandler: " << toAscii(box_type)
-        << Internal::stringFormat(" %8ld->%lu ", address, box_length);
+        << Internal::stringFormat(" %8ld->%llu ", address, box_length);
```
Comment (kmilos) [inline src/bmffimage.cpp:187]: Even better, let's use the [<cinttypes> macros](https://www.cplusplus.com/reference/cinttypes/) and forget portability issues popping back here.

Comment (piponazo) [reply src/bmffimage.cpp:187]: I'll give it a try for the first time in my life

File: src/bmffimage.cpp [hunk 2]
```diff
@@ -184,7 +185,7 @@ long BmffImage::boxHandler(std::ostream& out /* = std::cout*/, Exiv2::PrintStruc
   if (bTrace) {
     bLF = true;
     out << indent(depth) << "Exiv2::BmffImage::boxHandler: " << toAscii(box_type)
-        << Internal::stringFormat(" %8ld->%lu ", address, box_length);
+        << Internal::stringFormat(" %8" PRIi32 "->%" PRIu64 " ", address, box_length);
```
Comment (kmilos) [inline src/bmffimage.cpp:188]: 'address' is actually a long and not int32_t, so could/should leave it as %ld

Comment (piponazo) [reply src/bmffimage.cpp:188]: yeah ... I initially assumed that it was a `uint32_t` (by mistake) and I saw the LGTM error. Then I tried with `int32_t` but this is not the same as `long` ... In the [cinttypes](https://en.cppreference.com/w/cpp/header/cinttypes) doc, it does not seem to be anything to be used with the `long` type. I am surprised by this ...

I will leave it as `%8ld` for the moment.

File: src/tags.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ namespace Exiv2 {
 using namespace Internal;
 
 //! List of all defined Exif sections.
-constexpr auto sectionInfo = std::array{
+constexpr std::array<SectionInfo, 22> sectionInfo{
```
Comment (neheb) [inline src/tags.cpp:22]: What's the error here?

Comment (piponazo) [reply src/tags.cpp:22]: I'll check again tomorrow with the raspberry pi and copy&paste the error here.

Comment (piponazo) [reply src/tags.cpp:22]: Strange ... I dropped the commit and everything compiles now correctly. I might have get confused between the output from the compiler and the errors appearing in an old version of QtCreator. 

I will drop this commit from the PR.

**Review by kmilos — APPROVED** [review]
LGTM

**Comment by neheb** [comment]
clang-tidy now complains:

```
--- a/src/datasets.cpp
+++ b/src/datasets.cpp
@@ -508,7 +508,7 @@ IptcKey::IptcKey(uint16_t tag, uint16_t record) : tag_(tag), record_(record) {
   makeKey();
 }

-IptcKey::IptcKey(const IptcKey& rhs) : Key(), tag_(rhs.tag_), record_(rhs.record_), key_(rhs.key_) {
+IptcKey::IptcKey(const IptcKey& rhs) :  tag_(rhs.tag_), record_(rhs.record_), key_(rhs.key_) {
 }

 std::string IptcKey::key() const {
diff --git a/src/properties.cpp b/src/properties.cpp
index ebe089b8..707598b1 100644
--- a/src/properties.cpp
+++ b/src/properties.cpp
@@ -5097,7 +5097,7 @@ XmpKey::XmpKey(const std::string& prefix, const std::string& property) : p_(std:

 XmpKey::~XmpKey() = default;

-XmpKey::XmpKey(const XmpKey& rhs) : Key(), p_(std::make_unique<Impl>(*rhs.p_)) {
+XmpKey::XmpKey(const XmpKey& rhs) :  p_(std::make_unique<Impl>(*rhs.p_)) {
 }

 XmpKey& XmpKey::operator=(const XmpKey& rhs) {
diff --git a/src/tags.cpp b/src/tags.cpp
index 9cee60de..8fc02abf 100644
--- a/src/tags.cpp
+++ b/src/tags.cpp
@@ -266,7 +266,7 @@ ExifKey::ExifKey(const std::string& key) : p_(std::make_unique<Impl>()) {
   p_->decomposeKey(key);
 }

-ExifKey::ExifKey(const ExifKey& rhs) : Key(), p_(std::make_unique<Impl>(*rhs.p_)) {
+ExifKey::ExifKey(const ExifKey& rhs) :  p_(std::make_unique<Impl>(*rhs.p_)) {
 }

 ExifKey::~ExifKey() = default;

```

**Comment by piponazo** [comment]
Probably the issue is only appearing for the gcc-8.3 version ARM, but I am sure that this was giving me an error there. Could you share which cling-tidy rule is complaining about that? I could take a look and try to disable the particular warning on ARM if it makes sense.

**Comment by neheb** [comment]
https://clang.llvm.org/extra/clang-tidy/checks/readability-redundant-member-init.html

**Pull request merged**